### ISSUE TITLE
Unable to build the Image

### ISSUE THREAD

**Issue by dipjyotighosh** [oraclebase/dockerfiles#3]
Hi,

Please can you provide some inputs here. 
On looking into the logs the following issue shows up and the container dies.
For some reason it shows idle instance and unable to create files/directories.
This is for the first image itself :- o17_121

### **LSNRCTL for Linux: Version 12.1.0.2.0 - Production on 15-FEB-2019 05:54:39

Copyright (c) 1991, 2014, Oracle.  All rights reserved.

Starting /u01/app/oracle/product/12.1.0.2/db_1/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 12.1.0.2.0 - Production
System parameter file is /u01/app/oracle/product/12.1.0.2/db_1/network/admin/listener.ora
Log messages written to /u01/app/oracle/diag/tnslsnr/129a61f47f50/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC1)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 12.1.0.2.0 - Production
Start Date                15-FEB-2019 05:54:40
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Listener Parameter File   /u01/app/oracle/product/12.1.0.2/db_1/network/admin/listener.ora
Listener Log File         /u01/app/oracle/diag/tnslsnr/129a61f47f50/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
Look at the log file "/u01/app/oracle/cfgtoollogs/dbca/cdb1.log" for further details.

SQL*Plus: Release 12.1.0.2.0 Production on Fri Feb 15 05:55:10 2019

Copyright (c) 1982, 2014, Oracle.  All rights reserved.

Connected to an idle instance.

SQL> alter system set db_create_file_dest='/u02/oradata'
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0

SQL> Disconnected
mkdir: cannot create directory '/u02/config': Permission denied
cp: cannot create regular file '/u02/config/': Not a directory
sed: can't read /u02/config/oratab: No such file or directory
cp: cannot stat '/u02/config/oratab': No such file or directory
mv: cannot stat '/u01/app/oracle/product/12.1.0.2/db_1/dbs/orapwcdb1': No such file or directory
mv: cannot stat '/u01/app/oracle/product/12.1.0.2/db_1/dbs/spfilecdb1.ora': No such file or directory
mv: cannot move '/u01/app/oracle/admin' to '/u02/config/cdb1': No such file or directory
mv: cannot move '/u01/app/oracle/fast_recovery_area' to '/u02/config/cdb1': No such file or directory
mv: cannot move '/u01/app/oracle/diag' to '/u02/config/cdb1': No such file or directory
cp: cannot stat '/u02/config/oratab': No such file or directory

SQL*Plus: Release 12.1.0.2.0 Production on Fri Feb 15 05:55:12 2019

Copyright (c) 1982, 2014, Oracle.  All rights reserved.

Connected to an idle instance.

SQL> declare
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0

Disconnected
tail: cannot open '/u01/app/oracle/diag/rdbms/cdb1/cdb1/trace/alert_cdb1.log' for reading: No such file or directory
tail: no files remaining**

**Comment by oraclebase** [comment]
It looks like your container can't write to the external volume. I've added something into this article, which I think you are following, judging by and external comment.

http://oracle-base.com/articles/linux/docker-upgrade-an-oracle-databases-on-docker.php

It explains how to make the location visible to the container and a non-root user. I've also written about this here.

https://oracle-base.com/articles/linux/docker-host-file-system-permissions-for-container-persistent-host-volumes

I've made a number of changes to my older builds, like o17_121, o17_122 and o17_121_122, including support for the docker_fg group.

I'm working through the upgrade process again. There have been a lot of changes since these images were originally written.

**Comment by oraclebase** [comment]
I've run through the whole process now and it's working again.

Along the way I've made a lot of changes to all the builds. Some because I do things differently now. Some because external things have changed since these builds were first created.

You'll have to pull down a fresh copy of the repo before you try again, and make sure the file system is set up correctly.

If you are using my Vagrant Docker build, I've altered the setup somewhat for that also.
